server/src/main/java/org/elasticsearch/common/util/iterable/Iterables.java

```diff
@@ -35,11 +35,7 @@
 
         @Override
         public Iterator<T> iterator() {
-            return Stream.of(inputs)
-                .map(it -> StreamSupport.stream(it.spliterator(), false))
-                .reduce(Stream::concat)
-                .orElseGet(Stream::empty)
-                .iterator();
+            return Stream.of(inputs).flatMap(it -> StreamSupport.stream(it.spliterator(), false)).iterator();
```

Comment:
I looked into the uses of this method (and concatenating iterables in general). It appears there's literally only 3 spots and we already have concatenated iterators. Maybe we could just do away with the whole concept of concatenated iterable outright and replace the 3 use cases with either concatenating iterators or just doing some other simple thing? :)

Reply:
Correct. I would like to merge this as is (simplify concatenation in various places) and then open a followup pr to remove/replace operations from `org.elasticsearch.common.util.iterable.Iterables` that are no longer needed.